vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ML_PIPELINE_TEXT_PROCESSING_EMBEDDING_PROCESSING_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ML_PIPELINE_TEXT_PROCESSING_EMBEDDING_PROCESSING_H_
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "bat/ads/internal/ml/pipeline/pipeline_embedding_info.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_data.h"
+
+namespace base {
+class Value;
+}
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+struct EmbeddingPipelineInfo;
+
+class EmbeddingProcessing final {
+ public:
+  static std::unique_ptr<EmbeddingProcessing> CreateFromValue(
+      base::Value resource_value,
+      std::string* error_message);
+
+  EmbeddingProcessing();
+  ~EmbeddingProcessing();
+  EmbeddingProcessing(const EmbeddingProcessing& pipeline) = delete;
+  EmbeddingProcessing& operator=(const EmbeddingProcessing& pipeline) = delete;
+
+  bool IsInitialized() const;
+
+  void SetIsInitialized(bool is_initialized);
```

Comment:
Does not need to be public

Reply:
moved to private

Reply:
This method has actually now been removed, in response to a later comment